code/game/objects/items/cards_ids.dm

```diff
@@ -68,13 +68,56 @@
 	righthand_file = 'icons/mob/inhands/equipment/idcards_righthand.dmi'
 	item_flags = NO_MAT_REDEMPTION | NOBLUDGEON
 	var/prox_check = TRUE //If the emag requires you to be in range
+	var/can_bluespace = TRUE //can this become a bluespace emag?
 
 /obj/item/card/emag/bluespace
 	name = "bluespace cryptographic sequencer"
 	desc = "It's a blue card with a magnetic strip attached to some circuitry. It appears to have some sort of transmitter attached to it."
 	color = rgb(40, 130, 255)
 	prox_check = FALSE
 
+/obj/item/card/emag/improvised
+	name = "improvised cryptographic sequencer"
+	desc = "It's a card with some junk circuitry strapped to it. It doesn't look like it would be reliable or fast due to shoddy construction, and needs to be manually recharged with uranium sheets."
+	icon_state = "emag_shitty"
+	can_bluespace = FALSE //oh god no
+	var/charges = 5 //how many times can we use the emag before needing to reload it?
+	var/max_charges = 5
+	
+/obj/item/card/emag/improvised/afterattack(atom/target, mob/user, proximity)
+	if(charges > 0)
+		if(!proximity && prox_check) //left in for badmins
+			return
+		if(do_after(user, rand(50, 100), src))
+			var/atom/A = target
```

Comment:
```suggestion
	if(charges > 0)
		if(!proximity && prox_check) //left in for badmins
			return
		if(do_after(user, rand(5, 10) SECONDS, src))
			var/atom/A = target
```
you could early return and target is already an atom, so i don't know why the var here